lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -37,8 +39,11 @@ def exists?
     script_content = ps_script_content('test')
     Puppet.debug "\n" + script_content
     output = powershell(powershell_args, script_content)
-    if ['true','false'].include?(output.to_s.strip.downcase)
-      check = (output.to_s.strip.downcase == 'true')
+    Puppet.debug "Dsc Resource returned: #{output}"
+    data = JSON.parse(output)
+    fail(data['errormessage']) if data['errormessage']
```

Comment:
If `errormessage` is always present as an empty string, I believe this will always evaluate to true.

That's why I think we likely need another value in the serialized info to track error state of the operation that is set to `true` or `false`